src/main/c/Objects/pyjmethod.c

```diff
@@ -186,6 +187,11 @@ int PyJMethod_CheckArguments(PyJMethodObject* method, JNIEnv *env,
         paramTypeId = get_jtype(env, paramType);
 
         match = pyarg_matches_jtype(env, param, paramType, paramTypeId);
+        if (match == 0 && varargs && paramTypeId == JARRAY_ID) {
```

Comment:
This if statement needs to be limited to just the varargs parameter. If there is a method that takes an array and varargs then this will match the component type to the array.

For example this java method
```java
public String varargs(String[] arr, Object... varargs);
```
Would match this python call.
```python
obj.varargs("String")
```
Even though the call will fail.